### PR TITLE
Jens/update vignette

### PR DESCRIPTION
**Pull request by JensWahl**
* Add examples of the plotting functionality 
* Remove some of the examples for different approach methods
* Update the main arguments section

### CODE REVIEW

**Review by martinju — CHANGES_REQUESTED** [review]
I think this is OK except that beeswarm should also have more observations to explain. 
Mention that you use all observations (including those used for training for illustrations purposes)
+ Line 355 test observations -> observations

File: vignettes/understanding_shapr.Rmd [hunk 1]
```diff
@@ -282,57 +339,38 @@ print(explanation$shapley_values)
 
 # Plot the resulting explanations for observations 1 and 6
 plot(explanation, plot_phi0 = FALSE, index_x_explain = c(1, 6))
-``` 
-
-The Gaussian approach is used as follows:
+```
+There are multiple plot options speficified by the `plot_type`-argument in `plot`. The `waterfall` option will show the changes in the prediction score due to each features contribution (their Shapley values):
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:343]:
```suggestion
There are multiple plot options speficified by the `plot_type` argument in `plot`. The `waterfall` option shows the changes in the prediction score due to each features contribution (their Shapley values): 
```

File: vignettes/understanding_shapr.Rmd [hunk 2]
```diff
@@ -282,57 +339,38 @@ print(explanation$shapley_values)
...
-  approach = "gaussian",
-  prediction_zero = p0
-)
-# Plot the resulting explanations for observations 1 and 6
-plot(explanation_gaussian, plot_phi0 = FALSE, index_x_explain = c(1, 6))
+plot(explanation, plot_type = "waterfall", index_x_explain = c(1, 6))
 ```
 
-The Gaussian copula approach is used as follows:
+This final two plot options, `"beeswarm"` and `"scatter"` can be useful when you have many observations that you want to explain. For the purpose of illustration we will explain the whole `airquality` dataset (including the training data) for these plot types.
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:349]:
```suggestion
The final two plot options, `"beeswarm"` and `"scatter"`, can be useful when you have many observations that you want to explain. For the purpose of illustration, we explain the whole `airquality` dataset (including the training data) for these plot types. 
```

File: vignettes/understanding_shapr.Rmd [hunk 3]
```diff
@@ -282,57 +339,38 @@ print(explanation$shapley_values)
...
-  prediction_zero = p0
-)
-# Plot the resulting explanations for observations 1 and 6
-plot(explanation_gaussian, plot_phi0 = FALSE, index_x_explain = c(1, 6))
+plot(explanation, plot_type = "waterfall", index_x_explain = c(1, 6))
 ```
 
-The Gaussian copula approach is used as follows:
+This final two plot options, `"beeswarm"` and `"scatter"` can be useful when you have many observations that you want to explain. For the purpose of illustration we will explain the whole `airquality` dataset (including the training data) for these plot types. 
+The `plot_type = "beeswarm"` summarises the distribution of the Shapley values along the x-axis for all the features. Each point gives the shapley value of a given instance, where the points are colored by the feature value of that instance:
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:350]:
```suggestion
The `plot_type = "beeswarm"` summarises the distribution of the Shapley values along the x-axis across all features. Each point gives the Shapley value of a given instance, where the points are colored by the feature value of that instance:
```

File: vignettes/understanding_shapr.Rmd [hunk 4]
```diff
@@ -282,57 +339,38 @@ print(explanation$shapley_values)
...
   prediction_zero = p0
 )
-# Plot the resulting explanations for observations 1 and 6, excluding
-# the no-covariate effect
-plot(explanation_copula, plot_phi0 = FALSE, index_x_explain = c(1, 6))
+plot(explanation, plot_type = "beeswarm")
 ```
 
-The conditional inference tree approach is used as follows:
+Lastly, the `plot_type = "scatter"` plots the feature values on the x-axis and Shapley values on the y-axis, as well as
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:364]:
```suggestion
Lastly, the `plot_type = "scatter"` plots the feature values on the x-axis and Shapley values on the y-axis, as well as
```

File: vignettes/understanding_shapr.Rmd [hunk 5]
```diff
@@ -282,57 +339,38 @@ print(explanation$shapley_values)
...
+(optionally) a background scatter_hist showing the distribution of the feature data. :
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:365]:
```suggestion
(optionally) a background scatter_hist showing the distribution of the feature data:
```

File: vignettes/understanding_shapr.Rmd [hunk 6]
```diff
@@ -376,7 +414,9 @@ explanation_cat <- explain(
 plot(explanation_cat, plot_phi0 = FALSE, index_x_explain = c(1, 6))
 ```
 
-We can specify parameters used to build the conditional inference trees using the following manner. Default values are based on @hothorn2006unbiased.  
+We can specify parameters used to build the conditional inference trees
+using the following manner. Default values are based on
```
Comment (aredelmeier) [inline vignettes/understanding_shapr.Rmd:418]:
```suggestion
in the following manner. Default values are based on
```

**Review by aredelmeier — APPROVED** [review]

**Pull request merged**